src/metabase/pulse/render/datetime.clj

```diff
@@ -11,33 +11,107 @@
-  (cond (str/blank? s) ""
-
-        (isa? (or (:effective_type col) (:base_type col)) :type/Time)
-        (t/format DateTimeFormatter/ISO_LOCAL_TIME (u.date/parse s timezone-id))
-
-        :else
-        (case (:unit col)
-          ;; these types have special formatting
-          :hour    (reformat-temporal-str timezone-id s "h a - MMM yyyy")
-          :week    (str "Week " (reformat-temporal-str timezone-id s "w - YYYY"))
-          :month   (reformat-temporal-str timezone-id s "MMMM yyyy")
-          :quarter (reformat-temporal-str timezone-id s "QQQ - yyyy")
-
-          ;; no special formatting here : return as ISO-8601
-          ;; TODO: probably shouldn't even be showing sparkline for x-of-y groupings?
-          (:year :hour-of-day :day-of-week :week-of-year :month-of-year)
-          s
-
-          ;; for everything else return in this format
-          (reformat-temporal-str timezone-id s "MMM d, yyyy"))))
+  ([timezone-id s col] (format-temporal-str timezone-id s col {}))
+  ([timezone-id s col col-viz-settings]
+   (let [{date-style :date_style
+          abbreviate :date_abbreviate
+          time-style :time_style} col-viz-settings]
+     (cond (str/blank? s) ""
+
+           (isa? (or (:effective_type col) (:base_type col)) :type/Time)
+           (t/format DateTimeFormatter/ISO_LOCAL_TIME (u.date/parse s timezone-id))
+
+           :else
+           (case (:unit col)
+             ;; these types have special formatting
+             :minute  (reformat-temporal-str timezone-id s
+                                             (str (or date-style "MMMM, yyyy") ", "
+                                                  (str/replace (or time-style "h:mm a") #"A" "a")))
+             :hour    (reformat-temporal-str timezone-id s
+                                             (str (or date-style "MMMM, yyyy") ", "
+                                                  (str/replace (or time-style "h a") #"A" "a")))
+             :day     (reformat-temporal-str timezone-id s (or date-style "EEEE, MMMM d, YYYY"))
+             :week    (str (tru "Week ") (reformat-temporal-str timezone-id s "w - YYYY"))
+             :month   (reformat-temporal-str timezone-id s (or date-style "MMMM, yyyy"))
+             :quarter (reformat-temporal-str timezone-id s "QQQ - yyyy")
+             :year    (reformat-temporal-str timezone-id s "YYYY")
+
+             :day-of-week     (day-of-week s abbreviate) ;; s is just a number as a string here
+             :month-of-year   (month-of-year s abbreviate)
+             :quarter-of-year (format "Q%s" s)
+             :hour-of-day     (hour-of-day s (str/replace (or time-style "h a") #"A" "a"))
+
+             (:week-of-year :minute-of-hour :day-of-month :day-of-year) (x-of-y s)
+             ;; TODO: probably shouldn't even be showing sparkline for x-of-y groupings?
+
+             ;; for everything else return in this format
+             (reformat-temporal-str timezone-id s "MMM d, yyyy"))))))
```

Comment:
I've re-worked this function somewhat to optionally use `viz-settings`. I did this because prior, only column :unit data was used to switch cases, but the column unit is not sufficient. User selected viz-settings are also required (eg. 'abbreviate' boolean changes how Month names should be rendered).

While I was here, I changed the 'x-of-y' behaviours too, because we allow a user to make changes in the UI for those, we should (try) to respect those settings in static-viz too.

One thing I don't know about is if I have done the right thing handling, for example, :day-of-week. The string `s` being passed in that case is actually something like "2" (which means "Tuesday", regardless of the app's start-of-week setting). It's in the same category of 'temporal' rendering, but it is NOT actually an ISO-8601 string in this case. That's also true for `:month-of-year`, `:quarter-of-year`, `:hour-of-day`.